class.jetpack-client.php

```diff
@@ -299,11 +299,7 @@ static function wpcom_json_api_request_as_blog( $path, $version = self::WPCOM_JS
 		$_path = preg_replace( '/^\//', '', $path );
 
 		// Use GET by default whereas `remote_request` uses POST
-		if ( isset( $filtered_args['method'] ) && strtoupper( $filtered_args['method'] === 'POST' ) ) {
```

Comment:
Interestingly, the only reason has worked in the past is because `strtoupper(false)` evals to `""`, which is falsy in PHP, and `strtoupper(true)` evals to `"1"`, which is truthy. It would never have worked with `"post"` though.